src/Common/src/System/Globalization/IdnMapping.Unix.cs

```diff
@@ -82,11 +82,11 @@ private uint Flags
         /// To match Windows behavior, we walk the string ourselves looking for these
         /// bad characters so we can continue to throw ArgumentException in these cases. 
         /// </summary>
-        private static void CheckInvalidIdnCharacters(string s, uint flags, string paramName)
+        private static unsafe void CheckInvalidIdnCharacters(char* s, int startIndex, int count, uint flags, string paramName)
         {
             if ((flags & Interop.GlobalizationNative.UseStd3AsciiRules) == 0)
             {
-                for (int i = 0; i < s.Length; i++)
+                for (int i = startIndex; i < count; i++)
```

Comment:
There are tests for this.
I must admit my old laptop running Ubuntu is having problems with .NET core and I was waiting to see if the CI would pass on Unix